apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -375,6 +375,60 @@ export const commands = {
     return false;
   },
 
+  // Returns true if a minimum number of sprites are in the specified group.
+  minimumGroupSprites(group, min = 1) {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => this.nativeSpriteMap[id].group === group
+    );
+    return groupSpriteIds.length >= min;
+  },
+
+  // Returns true if a minimum number of sprites have no group.
+  minimumNonGroupSprites(min = 1) {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => !this.nativeSpriteMap[id].group
+    );
+    return groupSpriteIds.length >= min;
+  },
+
+  // Returns true if sprites with a minimum number of costumes are in the specified group.
+  minimumCostumesForGroup(group, min = 1) {
+    const uniqueCostumes = [];
+    const spriteIds = this.getSpriteIdsInUse();
+    const spritesInUse = spriteIds.map(id => this.nativeSpriteMap[id]);
+    const groupSprites = spritesInUse.filter(sprite => sprite.group === group);
+
+    for (const sprite of groupSprites) {
+      const costume = sprite.getAnimationLabel();
+      if (!uniqueCostumes.includes(costume)) {
+        uniqueCostumes.push(costume);
+      }
+    }
+    return uniqueCostumes.length >= min;
+  },
+
+  // Returns true if there is exactly one sprite with the "players" group.
+  playerSpriteFound() {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => this.nativeSpriteMap[id].group === 'players'
+    );
+    return groupSpriteIds.length === 1;
+  },
+
+  // Returns true if a player sprite has an upward vertical velocity
+  playerSpriteJumping() {
+    const spriteIds = this.getSpriteIdsInUse();
+    const spritesInUse = spriteIds.map(id => this.nativeSpriteMap[id]);
+
+    const jumpingSprites = spritesInUse.filter(
+      sprite => sprite.group === 'players' && sprite.velocityY < 0
```

Comment:
Is this supposed to be `< 0` and not `> 0`?

Reply:
No it’s correct as is. The velocity is negative if the sprite is moving up. This based on the actual P5 orientation and not a Sprite Lab block that remaps it in the other direction.